### PR TITLE
watcher: switch to algo indexer

### PR DESCRIPTION
**Pull request by evan-gray**

### CODE REVIEW

File: watcher/src/watchers/AlgorandWatcher.ts [hunk 1]
```diff
@@ -290,62 +30,90 @@ export class AlgorandWatcher extends Watcher {
...
-        }
+  async getApplicationLogTransactionIds(fromBlock: number, toBlock: number): Promise<string[]> {
+    // it is possible tihs may result in gaps if toBlock > response['current-round']
+    // perhaps to avoid this, getFinalizedBlockNumber could use the indexer?
+    let transactions: string[] = [];
+    let nextToken: string | undefined;
+    let numResults: number | undefined;
+    const maxResults = 225; // determined through testing
+    do {
+      const request = await this.indexerClient
```
Comment (panoel) [inline watcher/src/watchers/AlgorandWatcher.ts:48]: I don't think there are any promises involved in creating the request.

File: watcher/src/watchers/AlgorandWatcher.ts [hunk 2]
```diff
@@ -1,280 +1,20 @@
...
-  }
-}
+type Message = {
+  blockKey: string;
+  vaaKey: string;
+};
 
 export class AlgorandWatcher extends Watcher {
-  maximumBatchSize: number = 5;
+  maximumBatchSize: number = 100000;
```
Comment (panoel) [inline watcher/src/watchers/AlgorandWatcher.ts:13]: This is just set arbitrarily large, right?  Just limits the size of the string[] being returned?

**Review by panoel — DISMISSED** [review]

File: watcher/src/watchers/AlgorandWatcher.ts (hunk 2 above)
Comment (evan-gray) [reply watcher/src/watchers/AlgorandWatcher.ts:13]: Left a comment - yes it's just arbitrarily large. Has no impact on the return either, just the largest range the `watch` loop will pass in.

File: watcher/src/watchers/AlgorandWatcher.ts [hunk 3]
```diff
@@ -290,62 +30,90 @@ export class AlgorandWatcher extends Watcher {
...
-            ret.push(obs);
-          }
-          for (const txn of eval_delta['inner_txns']) {
-            ret = ret.concat(this.scanTxn(txn, rnd));
-          }
-        }
+  async getApplicationLogTransactionIds(fromBlock: number, toBlock: number): Promise<string[]> {
+    // it is possible tihs may result in gaps if toBlock > response['current-round']
+    // perhaps to avoid this, getFinalizedBlockNumber could use the indexer?
+    let transactions: string[] = [];
```
[the remark this answers is not in the document]
Comment (evan-gray) [reply watcher/src/watchers/AlgorandWatcher.ts:43]: Absolutely, thanks!

File: watcher/src/watchers/AlgorandWatcher.ts (hunk 1 above)
Comment (evan-gray) [reply watcher/src/watchers/AlgorandWatcher.ts:48]: Thanks :grin: this is what I get for moving the code around after I wrote it.

**Review by panoel — APPROVED** [review]

**Review by kev1n-peters — APPROVED** [review]

**Pull request merged**